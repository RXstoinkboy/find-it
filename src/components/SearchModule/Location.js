import React, {Component} from 'react';
import styles from './styles/Location.module.scss'
import utils from './styles/SearchInput.module.scss'
import PropTypes from 'prop-types'

import CitiesMod from './CitiesMod'

export class Location extends Component {
    constructor(props){
        super(props);
        this.state = {
            focus: false
        }
    }

    toggleFocus = e => {
        this.setState({
            focus: !this.state.focus
        })
    }

    render(){
        return (
            <div className={utils.wrapper}>
                <input 
                    type="search" 
                    placeholder='Where are you?' 
                    onChange={this.props.handleSetCity}
                    onFocus={this.toggleFocus}
                    onBlur={this.toggleFocus}
                    className={styles.location} 
                    value={this.props.city}/>
                <CitiesMod {...this.props} focus={this.state.focus} />
            </div>
        );
    }
};

export default Location;

Location.propTypes = {
    handleSetCity: PropTypes.func.isRequired,
    city: PropTypes.string.isRequired,
    cities: PropTypes.array.isRequired
}