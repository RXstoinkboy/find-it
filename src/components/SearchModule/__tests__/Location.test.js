import React from 'react'
import {Location} from '../Location'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

it(`Location renders correctly`, ()=> {
    const mockFn = jest.fn();
    const location = shallow(<Location city='New York' handleCity={mockFn}/>)

    expect(toJson(location)).toMatchSnapshot()
})