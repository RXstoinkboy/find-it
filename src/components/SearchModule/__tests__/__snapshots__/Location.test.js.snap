// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location renders correctly 1`] = `
<div
  className="wrapper"
>
  <input
    className="location"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    placeholder="Where are you?"
    type="search"
    value="New York"
  />
  <CitiesMod
    cities={
      Array [
        1,
        2,
      ]
    }
    city="New York"
    focus={false}
    handleSetCity={[MockFunction]}
  />
</div>
`;
